api4/context.go

```diff
@@ -375,6 +375,38 @@ func (c *Context) RequireUsername() *Context {
 	return c
 }
 
+func (c *Context) RequireTeamName() *Context {
+	if c.Err != nil {
+		return c
+	}
+
+	if len(c.Params.TeamName) < 2 {
+		c.SetInvalidUrlParam("team_name")
+	} 
+
+	if len(c.Params.TeamName) > 15 {
+		c.SetInvalidUrlParam("team_name")
+	}
+
+	return c
+}
+
+func (c *Context) RequireChannelName() *Context {
+	if c.Err != nil {
+		return c
+	}
+
+	if len(c.Params.ChannelName) < 2 {
+		c.SetInvalidUrlParam("channel_name")
+	} 
+
+	if len(c.Params.ChannelName) > 22 {
```

Comment:
I found it the constants for teams [here](https://github.com/mattermost/platform/blob/master/webapp/utils/constants.jsx#L856) and the channels [here](https://github.com/mattermost/platform/blob/master/webapp/components/rename_channel_modal.jsx#L21) and didn't understand if `model.CHANNEL_NAME_MAX_LENGTH ` had the same functionality. I'll update it! 👍